
# Author: Alicia Garcia-Roldan

# Date: September 2023

# R version 4.1.2


# Drawing a bubble plot to show the recruitment results
# We can use the table generated by modifying the results of the recruitment on Excel


## LIBRARY ##

library(paletteer)  # v.1.4.0
library(ggplot2)  # v.3.4.0


## DATASET ##

allrec <- c(IC195 = "Reclutamientos_IC19_5.txt", IC22="Reclutamientos_IC22.txt",
            IC36_1="Reclutamientos_IC361.txt", IC36_2="Reclutamientos_IC362.txt",
            IC39_1="Reclutamientos_IC391.txt", IC39_2="Reclutamientos_IC392.txt",
            IC21="Reclutamientos_IC21.txt", 
            SS13= "Reclutamientos_SS13.txt", SS19="Reclutamientos_SS19.txt",
            SS33="Reclutamientos_SS33.txt", SS37="Reclutamientos_SS37.txt",
            Meyghan_5 = "Reclutamientos_Meyghan5_nuevo.txt", Meyghan_18="Reclutamientos_Meyghan18_nuevo.txt",
            Meyghan_30="Reclutamientos_Meyghan30_nuevo.txt",
            ArcticSpring = "Reclutamientos_Arctic24_nuevo.txt", CaboRojo="Reclutamientos_CaboRojo34_nuevo.txt",
            Urmia="Reclutamientos_Urmia27_nuevo.txt", Xinjiang="Reclutamientos_Xinjiang30_nuevo.txt",
            d1A="Reclutamientos_d1A_nuevo.txt")

projs=list()

for (sample in names(allrec)) {
  projs[[sample]] = read.csv(allrec[[sample]], header=T, sep="\t", quote = "")
  projs[[sample]] <- cbind(projs[[sample]], c(rep(sample, nrow(projs[[sample]])))) # Add sample column, easier
                                                                                   # to identify
  colnames(projs[[sample]]) <- c("Genome", "Reads", "Genoma (bp)", "Genoma (Kb)",
                                 "Metagenoma (bp)", "Metagenoma (Kb)", "RPKG", "Sample") # Renaming my table :)
}

df <- data.frame(matrix(ncol = 8, nrow = 0))

for (sample in names(projs)){
  df <- rbind(df, projs[[sample]])
}


## PLOTTING TIME! ##

# Barplot changes your order, so if we want the correct order we have to force it :(

ordenx <- c("IC195", "IC22","IC36_1", "IC36_2", "IC39_1", "IC39_2",
            "IC21", 
            "SS13", "SS19", "SS33", "SS37",
            "Meyghan_5", "Meyghan_18", "Meyghan_30",
            "ArcticSpring", "CaboRojo", "Urmia", "Xinjiang",
            "d1A")

ordeny <- c("c157", "c78", "c65", "c104",
            "Spiribacter_salinus_M1940", "Haloquadratum_walsbyi_C23")

colores <- c("#B40F20", "#E58601", rep("#46ACC8",2), rep("#E2D200",2), rep("grey80", 13))  # Thanks again, Wes Anderson

ggplot(df, aes(x = Sample, y = Genome, size = RPKG, fill = RPKG)) + 
  geom_point(shape = 21, alpha=0.8, stroke = 1.1, colour = "black") +
  theme(legend.position = "right",             # Change legend position
        legend.key.size = unit(0.5, 'cm'),     # Change legend key size
        legend.text = element_text(size=11),   # Change legend text font size
        legend.title = element_text(size=13.5, face="bold"),
        legend.key=element_blank(),
        
        axis.ticks.x = element_blank(),        # Delete the little lines marking the values
        axis.text.x = element_text(colour = "black",
                                   size = 11,
                                   face = "bold",
                                   angle = 45,
                                   vjust = 1,
                                   hjust = 1),
        axis.title.x = element_blank(),
        axis.ticks.y = element_blank(),        # Delete the little lines marking the values
        axis.text.y = element_text(colour = "black",
                                   face = "bold",
                                   size = 11),
        axis.title.y = element_blank(),
        
        panel.background = element_blank(), panel.border = element_rect(colour = "black", fill = NA, size = 1.2),
        panel.grid.major.x = element_line(size = 0.5, linetype =2, # Longest lines that go through the plot
                                          colour = colores)
        ) +
  
  scale_y_discrete(limits = ordeny) +
  scale_x_discrete(limits = ordenx) +
  scale_fill_gradientn (colours =
                        paletteer_d("vapoRwave::vapoRwave"),   # My second best friend: Vaporwave art
                        values = scales::rescale(c(0, 1, 2, 3, 4, 5, 10, 15, 20, 30, 40, 50,
                                                   60, 70, 80, 90, 100,
                                                   110, 120, 130, 140, 150, 160, 170, 180, 190, 200,
                                                   210, 220, 230, 240, 250, 260, 270, 270, 280, 290, 300)))+
  scale_size(limits = c(0, 307), range= c(4,20), breaks = c(0, 20, 50, 80, 100, 200, 300))


## SAVE ## 

ggsave("Recruitment_bubble.png", width = 30, height = 20, units = "cm", bg = "transparent")
ggsave("Recruitment_bubble.pdf", width = 30, height = 20, units = "cm", bg = "transparent")  
  
